src/makeCalculator.js

```diff
@@ -37,7 +37,41 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const INITIAL_VALUE = 0;
+
+  const calculator = {
+    add(integer) {
+      calculator.result += integer;
+    },
+
+    subtract(integer) {
+      calculator.result -= integer;
+    },
+
+    multiply(integer) {
+      calculator.result *= integer;
+    },
+
+    divide(integer) {
+      calculator.result /= integer;
+    },
+
+    reset(integer) {
+      calculator.result = INITIAL_VALUE;
+
+      return this;
+    },
+
+    operate(action, integer) {
+      action(integer);
```

Comment:
Recommendation:

we can use `bind`, `call` or `apply`  to have proper `this` inside our callback.



[call-apply-and-bind](https://www.freecodecamp.org/news/understand-call-apply-and-bind-in-javascript-with-examples/)